modules/c++/re/unittests/test_regex.cpp

```diff
@@ -0,0 +1,312 @@
+/* =========================================================================
+ * This file is part of re-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * re-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <import/re.h>
+#include "TestCase.h"
+#include <map>
+
+TEST_CASE(testCompile)
+{
+    re::Regex rx;
+    // test that an invalid regexp throws an exception
+    TEST_EXCEPTION(rx.compile("^("));
+
+    // test that a valid regexp compiles
+    try
+    {
+        rx.compile("^(foo)");
+    }
+    catch (...)
+    {
+        TEST_FAIL("Compiling a valid regexp should not have thrown exception!");
+    }
+}
+
+TEST_CASE(testMatches)
+{
+    re::RegexMatch matches;
+    re::Regex rx("^([^:]+):[ ]*([^\r\n]+)\r\n(.*)");
+    TEST_ASSERT(rx.match("Proxy-Connection: Keep-Alive\r\n", matches));
+    TEST_ASSERT_EQ(matches.size(), 4);
+}
+
+TEST_CASE(testSearch)
+{
+    re::Regex rx("jud");
+    std::string result = rx.search("arabsdsarbjudarc34ardnjfsdveqvare3arfarg");
+    TEST_ASSERT_EQ(result, "jud");
+}
+
+TEST_CASE(testSearchAll)
+{
+    re::RegexMatch matches;
+    re::Regex rx("ar");
+    rx.searchAll("arabsdsarbjudarc34ardnjfsdveqvare3arfarg", matches);
+    TEST_ASSERT_EQ(matches.size(), 7);
+}
+
+TEST_CASE(testDotAllFlag)
+{
+    // This should match both the "3.3" and "4\n2"
+    re::RegexMatch matches1;
+    re::Regex rx1("\\d.\\d");
+    rx1.searchAll("3.3 4\n2", matches1);
+    TEST_ASSERT_EQ(matches1.size(), 2);
+
+    // This should only match the "3.3"
```

Comment:
Yeah, that test was from before I removed the public DOTALL flag.  I removed the commented-out code.